bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/components/UIComponentSitemapProvider.java

```diff
@@ -261,8 +273,8 @@ protected Widget buildWidget(UIComponent component) {
         return widget;
     }
 
-    private void setWidgetPropertyFromComponentConfig(Widget widget, UIComponent component, String configParamName,
-            int feature) {
+    private void setWidgetPropertyFromComponentConfig(Widget widget, @Nullable UIComponent component,
```

Comment:
Something beyond scope of this PR but passing a `@Nullable UIComponent component` here seems to be wrong. Could be captured by calling methods.

Reply:
I was still unsure, after a short exploration of the call tree, if it would actually happen because of the null check in this method. So I opted for the conservative route of adding the annotation.